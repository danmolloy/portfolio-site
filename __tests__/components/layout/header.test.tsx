import Header, { HeaderProps, menuItems } from "@/components/layout/header";
import { act, fireEvent, render, screen } from '@testing-library/react'
import { animateScroll } from 'react-scroll'

jest.mock("react-scroll", () => ({
  animateScroll: {
    scrollToTop: jest.fn()
  },
  Link: (props: any) => <div></div>
}))

const mockProps: HeaderProps = {
  showMenu: false,
  setShowMenu: jest.fn(),
  setShowHeader: jest.fn(),
  showHeader: true
}

describe("<Header />", () => {
  beforeEach(() => {
    render(<Header {...mockProps} />)
  })
  it("header is in the document", () => {
    const header = screen.getByTestId("header")
    expect(header).toBeInTheDocument()
  })
  it('home btn is in the document and calls scrollToTop on click', () => {
    const homeBtn = screen.getByTestId("home-btn")
    expect(homeBtn).toBeInTheDocument()
    expect(homeBtn.textContent).toMatch("Home")
    act(() => {
      fireEvent.click(homeBtn)
    })
    expect(animateScroll.scrollToTop).toHaveBeenCalledWith({
      smooth: true, duration: 500
    })
  })
  it("menu icon is in the document and calls showMenu on click", () => {
    const menuIcon = screen.getByTestId("menu-icon")
    expect(menuIcon).toBeInTheDocument()
    act(() => {
      fireEvent.click(menuIcon)
    })
    expect(mockProps.setShowMenu).toHaveBeenCalled()
  })
  it("matches snapshot", () => {
    const header = screen.getByTestId("header")
    expect(header).toMatchSnapshot()
  })
  //it("all menuItems are in the document with expected to attrs and text content", () => {})
})