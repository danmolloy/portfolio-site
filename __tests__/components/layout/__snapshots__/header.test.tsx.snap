// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<div
  class="justify-between font-display transition duration-500 ease-out fixed  bg-white z-30 h-14 w-screen top-0 flex flex-row-reverse md:flex-row  items-center  shadow fill-zinc-300 text-black"
  data-testid="header"
>
  <div
    class="hidden md:flex w-screen  flex-row  justify-end pr-4"
  >
    <button
      class=" p-4 m-2 text-lg font-light hover:cursor-pointer active:text-zinc-400"
      data-testid="home-btn"
    >
      Home
    </button>
    <div />
    <div />
    <div />
  </div>
  <button
    class="fill-black m-2 hover:fill-orange-500 active:fill-orange-400  md:hidden "
    data-testid="menu-icon"
  >
    <svg
      class="m-2"
      height="22"
      width="30"
    >
      <rect
        height="2"
        rx="2"
        style="transform: scaleX(1) translate(0px, 0px) rotate(0deg);"
        width="30"
      />
      <rect
        height="2"
        rx="2"
        style="opacity: 1;"
        width="30"
        y="8"
      />
      <rect
        height="2"
        rx="2"
        style="transform: scaleX(1) translate(0px, 0px) rotate(0deg);"
        width="30"
        y="17"
      />
    </svg>
  </button>
</div>
`;
